resources/data_structures_algorithms.md

```diff
@@ -0,0 +1,255 @@
+# Data Structures and Algorithms
+
+---
+
+<details markdown="1">
+  <summary>Table of Contents</summary>
+
+-
+
+</details>
+
+---
+
+# Intro to Algorithms
```

Comment:
Could use a high level introduction to algorithms here. For example, what is an algorithm and what is the goal of a good algorithm?